### PR TITLE
[pileup_interval] py2 -> py3

### PR DESCRIPTION
**Pull request by almahmoud**
Came up in containerized tool tests. This makes it run. Alternative is setting py 2 requirement.

### CODE REVIEW

**Comment by bgruening** [comment]
@almahmoud can you please bump the tool version.

**Comment by mvdbeek** [comment]
Can you also remove this from `https://github.com/galaxyproject/tools-devteam/blob/master/.tt_skip#L67` and add a requirement for `bx-python` ?

**Comment by almahmoud** [comment]
Done

**Comment by mvdbeek** [comment]
Nice! Can you also fix the linting errors ?

**Comment by bgruening** [comment]
@almahmoud ping

**Comment by almahmoud** [comment]
Done, although just for the record I think all of the linting issues were unrelated to the small py2->py3 change

File: tools/pileup_interval/pileup_interval.py [hunk 1]
```diff
@@ -14,23 +14,26 @@
    -r, --base_column=r: Reference base column
    -C, --cvrg_column=C: Coverage column
 """
+import sys
 
-from galaxy import eggs
-import pkg_resources; pkg_resources.require( "bx-python" )
+import pkg_resources
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.py:19]:
```suggestion
```

File: tools/pileup_interval/pileup_interval.py [hunk 2]
```diff
@@ -59,20 +62,20 @@ def __main__():
         lineParts = inLine.split('\t')
         try:
             seq, loc, base, cov = lineParts[seqIndex], int(lineParts[locIndex]), lineParts[baseIndex], int(lineParts[covIndex])
-        except IndexError, ei:
+        except IndexError as ei:
             if options.format == 'ten':
-                stop_err( 'It appears that you have selected 10 columns while your file has 6. Make sure that the number of columns you specify matches the number in your file.\n' + str( ei ) )
+                stop_err('It appears that you have selected 10 columns while your file has 6. Make sure that the number of columns you specify matches the number in your file.\n' + str(ei))
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.py:67]:
```suggestion
                sys.exit('It appears that you have selected 10 columns while your file has 6. Make sure that the number of columns you specify matches the number in your file.\n' + str(ei))
```

File: tools/pileup_interval/pileup_interval.py [hunk 3]
```diff
@@ -59,20 +62,20 @@ def __main__():
...
             else:
-                stop_err( 'There appears to be something wrong with your column index values.\n' + str( ei ) )
-        except ValueError, ev:
+                stop_err('There appears to be something wrong with your column index values.\n' + str(ei))
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.py:69]:
```suggestion
                sys.exit('There appears to be something wrong with your column index values.\n' + str(ei))
```

File: tools/pileup_interval/pileup_interval.py [hunk 4]
```diff
@@ -59,20 +62,20 @@ def __main__():
...
+        except ValueError as ev:
             if options.format == 'six':
-                stop_err( 'It appears that you have selected 6 columns while your file has 10. Make sure that the number of columns you specify matches the number in your file.\n' + str( ev ) )
+                stop_err('It appears that you have selected 6 columns while your file has 10. Make sure that the number of columns you specify matches the number in your file.\n' + str(ev))
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.py:72]:
```suggestion
                sys.exit('It appears that you have selected 6 columns while your file has 10. Make sure that the number of columns you specify matches the number in your file.\n' + str(ev))
```

File: tools/pileup_interval/pileup_interval.py [hunk 5]
```diff
@@ -59,20 +62,20 @@ def __main__():
...
             else:
-                stop_err( 'There appears to be something wrong with your column index values.\n' + str( ev ) )
+                stop_err('There appears to be something wrong with your column index values.\n' + str(ev))
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.py:74]:
```suggestion
                sys.exit('There appears to be something wrong with your column index values.\n' + str(ev))
```

File: tools/pileup_interval/pileup_interval.py [hunk 6]
```diff
@@ -14,23 +14,26 @@
...
-    sys.stderr.write( msg )
+pkg_resources.require("bx-python")
+
+
+def stop_err(msg):
+    sys.stderr.write(msg)
     sys.exit()
 
+
 def __main__():
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.py:30]:
```suggestion

def __main__():
```

File: tools/pileup_interval/pileup_interval.py [hunk 7]
```diff
@@ -14,23 +14,26 @@
...
-    strout = ''
-    #Parse Command Line
-    options, args = doc_optparse.parse( __doc__ )
+    # strout = ''
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.py:31]:
```suggestion
```

File: tools/pileup_interval/pileup_interval.xml [hunk 8]
```diff
@@ -1,4 +1,7 @@
-<tool id="pileup_interval" name="Pileup-to-Interval" version="1.0.2">
+<tool id="pileup_interval" name="Pileup-to-Interval" version="1.0.3">
+  <requirements>
+    <requirement type="package" version="0.7.1">bx-python</requirement>
+  </requirements>
   <description>condenses pileup format into ranges of bases</description>
   <command interpreter="python">
     pileup_interval.py
```
Comment (nsoranzo) [inline tools/pileup_interval/pileup_interval.xml:7]:
```suggestion
  <command>
python '$__tool_directory__/pileup_interval.py'
```

**Review by bgruening — APPROVED** [review]

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**